l10n_es_dua_sii/models/account_invoice.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2017 Consultoría Informática Studio 73 S.L.
+# Copyright 2017 Comunitea Servicios Tecnológicos S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api, tools
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.model
+    @tools.ormcache('company')
+    def _get_dua_fiscal_position(self, company):
+        dua_fiscal_position = self.env.ref(
+            'l10n_es_dua.%i_fp_dua' % (company.id),
+            raise_if_not_found=False
+        ) or self.env['account.fiscal.position'].search([
+            ('name', '=', 'Importación con DUA')
+        ])
+        return dua_fiscal_position
+
+    @api.multi
+    @api.depends('tax_line_ids')
+    def _compute_sii_enabled(self):
+        super(AccountInvoice, self)._compute_sii_enabled()
+        for invoice in self.filtered('sii_enabled'):
+            dua_fiscal_position = self._get_dua_fiscal_position(
+                invoice.company_id)
+            if dua_fiscal_position and \
+                    invoice.fiscal_position_id == dua_fiscal_position and \
+                    not invoice.sii_dua_invoice:
+                invoice.sii_enabled = False
+
+    @api.multi
+    def _compute_dua_invoice(self):
+        taxes = self._get_sii_taxes_map(['DUA'])
```

Comment:
Esta línea debe estar dentro del `for` porque `_get_sii_taxes_map` espera una única factura:
```suggestion
        for invoice in self:
            taxes = invoice._get_sii_taxes_map(['DUA'])
```